src/main/java/org/osc/manager/rest/server/api/SecurityApis.java

```diff
@@ -0,0 +1,173 @@
+    @POST
+    public String createSecurityGroupInterface(@PathParam("sgId") Long sgId, SecurityGroupInterfaceEntity entity)
+            throws Exception {
+        LOG.info(String.format("Creating the security group interface with name %s ", "" + entity.getName()));
+        SecurityGroupEntity sgElement = new SecurityGroupEntity();
+        sgElement.setId(sgId);
+        entity.setSecurityGroup(sgElement);
+        return this.sgiApi.createSecurityGroupInterface(entity);
+    }
+
+    @Path("/{sgId}/sgIntf/{sgIntfId}")
+    @PUT
+    public SecurityGroupInterfaceElement updateSecurityGroupInterface(@PathParam("sgId") Long sgId,
+            @PathParam("sgIntfId") Long sgIntfId, SecurityGroupInterfaceEntity entity) throws Exception {
+        LOG.info(String.format("Updating the security group interface with sgid %s ; sginterfaceid %s",
+                "" + Long.toString(sgId), "" + Long.toString(sgIntfId)));
+        SecurityGroupEntity sgElement = new SecurityGroupEntity();
+        sgElement.setId(sgId);
+        entity.setSecurityGroup(sgElement);
+        entity.setId(sgIntfId);
+        this.sgiApi.updateSecurityGroupInterface(entity);
+        return entity;
+    }
+
+    @Path("/{sgId}/sgIntf/{sgIntfId}")
+    @DELETE
+    public void deleteSecurityGroupInterface(@PathParam("sgId") Long sgId, @PathParam("sgIntfId") Long sgIntfId)
+            throws Exception {
+        LOG.info(String.format("Deleting the security group interface with sgid %s ; sginterfaceid %s",
+                "" + Long.toString(sgId), "" + Long.toString(sgIntfId)));
+        this.txControl.required(new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                SecurityApis.this.sgiApi.deleteSecurityGroupInterface(Long.toString(sgIntfId));
+                return null;
+            }
+        });
+    }
+
+    @Path("/{sgId}/sgIntf")
+    @GET
+    public List<String> getSecurityGroupInterfaceIds() throws Exception {
+        LOG.info("Listing security group interface ids'");
+        List<? extends ManagerSecurityGroupInterfaceElement> sgiElements = this.sgiApi.listSecurityGroupInterfaces();
+        List<String> sgiList = new ArrayList<String>();
+        if (!sgiElements.isEmpty()) {
+            sgiList = sgiElements.stream().map(ManagerSecurityGroupInterfaceElement::getSecurityGroupInterfaceId)
+                    .collect(Collectors.toList());
+        }
+        return sgiList;
+    }
+
+    @Path("/{sgId}/sgIntf/{sgIntfId}")
+    @GET
+    public SecurityGroupInterfaceEntity getSecurityGroupInterface(@PathParam("sgId") Long sgId,
+            @PathParam("sgIntfId") Long sgIntfId) throws Exception {
+        LOG.info(String.format("getting the security group interface with sgid %s ; sginterfaceid %s)",
+                "" + Long.toString(sgId), "" + Long.toString(sgIntfId)));
+        return (SecurityGroupInterfaceEntity) this.sgiApi.getSecurityGroupInterfaceById(Long.toString(sgIntfId));
```

Comment:
Same comment here, you likely need to filter by parent (SG id) as well.

Reply:
OK